### ISSUE TITLE
OpenSSL library and header mismatch causing poor performance

### ISSUE THREAD

**Issue by haphan** [phpseclib/phpseclib#597]
https://github.com/phpseclib/phpseclib/blob/master/phpseclib/Math/BigInteger.php#L286-L294

phpseclib takes advantage of openssl feature only if OpenSSL library and OpenSSL header version are identical. 

Major distributions like Apple PHP and MAMP PHP were compiled with minor difference in version. 

Apple PHP
```bash
haphan@koala:~/$ /usr/bin/php -i | grep OpenSSL
OpenSSL support => enabled
OpenSSL Library Version => OpenSSL 0.9.8za 5 Jun 2014
OpenSSL Header Version => OpenSSL 0.9.8y 5 Feb 2013
OpenSSL support => enabled
```

MAMP
```bash
haphan@koala:~/$ php -i | grep OpenSSL
SSL Version => OpenSSL/0.9.8z
OpenSSL support => enabled
OpenSSL Library Version => OpenSSL 0.9.8za 5 Jun 2014
OpenSSL Header Version => OpenSSL 0.9.8r 8 Feb 2011
Native OpenSSL support => enabled
```

I think phpseclib should only check for major version of openSSL.

**Comment by terrafrost** [comment]
Overall I think it's a good commit but... 

```php
                            //Look for major version
                            $fullVersion = trim(str_replace('=>', '', strip_tags($matches[2][$i])));
                            preg_match('/(\d+\.\d+\.\d+)/i', $fullVersion, $m);

                            if (!empty($m)) {
                                $versions[$matches[1][$i]] = $m[0];
                            } else {
                                $versions[$matches[1][$i]] = $fullVersion;
                            }
```

You could do this instead:

```php
                            //Look for major version
                            $fullVersion = trim(str_replace('=>', '', strip_tags($matches[2][$i])));
                            if (!preg_match('/(\d+\.\d+\.\d+)/i', $fullVersion, $m)) {
                                $versions[$matches[1][$i]] = $fullVersion;
                            } else {
                                $versions[$matches[1][$i]] = $m[0];
                            }
```

Also, saying major version is a little misleading. With semantic versioning you have [major].[minor].[patch], so people used to versioning like that might think you're only checking to see if it's OpenSSL 1.0 headers with an OpenSSL 2.0 library. Maybe a better comment would be... "remove [letter versions](http://stackoverflow.com/a/10347301/569976)] or some such.

**Comment by terrafrost** [comment]
Fixed by #598
